options/cf_options.cc

```diff
@@ -299,6 +302,7 @@ static std::unordered_map<std::string, OptionTypeInfo>
               }
               return s;
             } else if (old_opts != nullptr) {
+              // In case of safely mutable subcomponents of the table factory
```

Comment:
If I call `SetOptions({{"block_based_table_factory.filter_policy", "ribbonfilter:6.67:3"}})`, it will hit this path and updates a share_ptr option in-place, which is not safe?

Reply:
Yeah, it was looking like it worked a certain way but that comment is not accurate. The Configurable framework is very awkward to work with and will need some extension to patch the usability / consistency holes.